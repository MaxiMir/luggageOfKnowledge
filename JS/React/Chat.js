/**
 * UI Frameworks:
 - https://ant.disign <-
 - https://material-ui.com
 - https://blueprintjs.com
 - https: semantic-ui.com
 - SASS / styled-components

 * JS библиотеки
 - Redux
 - Redux-thunk
 - Socket.io
 - axios
 - Lodash
 - Formik
 - React Route
 - classnames

 * Server
 - NodeJS (ExpressJS)
 - MongoDB / Mongoose
 - Multer (для загрузки файлов на сервер)
 - PassportJS (организация авторизации, в тч через соц. сети)



 $ create-react-app chat
 $ yarn start

 Удаляем:
 > FILE: /src/App.css
 > FILE: /src/App.test.js
 > FILE: /src/logo.svg
 > FILE: /src/serviceWorker.js

 $ yarn eject

 Frontend https://github.com/Archakov06/react-chat-tutorial
 Backend https://github.com/Archakov06/backend-chat-tutorial
 */


// FILE: /config/webpack.config.js:
/*
...
resolve: {
    modules: ['node_modules'].concat(
    ->
    modules: ['node_modules', 'src'].concat( // для иморта модулей
}
...



    $ yarn add node-sass
    $ yarn add classnames
    $ yarn start


 + FOLDER /src/components/              глобальные компоненты
    + FILE /src/components/index.js

 + FOLDER /src/modules/                 функциональные компоненты
 + FOLDER /src/actions/
 + FOLDER /src/reducers/
 + FOLDER /src/styles/
    + FILE /src/styles/index.scss
 */

// FILE: /src/index.js:
import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router } from "react-router-dom";

import App from './App';

import './styles/index.scss';

ReactDOM.render(
    <Router>
        <App/>
    </Router>,
    document.getElementById('root')
);


/*
    $ yarn add antd
 */


// FILE: /src/styles/index.scss:
/*
    @import "~antd/dist/antd.css"; // стили UI библиотеки ant.disign


*/


/*
    + FOLDER /src/components/Button/
        + FILE /src/components/Button/index.js
        + FILE /src/components/Button/Button.scss
 */


// FILE: /src/components/Button/index.js
import React from "react";
import PropTypes from "prop-types";
import { Button as BaseButton } from "antd";
import classNames from "classnames";

import "./Button.scss";

const Button = props => (
    <BaseButton
        {...props}
        className={classNames("button", props.className, {
            "button--large": props.size === "large"
        })}
    />
);

Button.propTypes = {
    className: PropTypes.string
};

export default Button;


// FILE /src/components/Button/Button.scss
/*
    .button {
        &--large {
            height: 56px !important;
        }
    }
 */


// FILE /src/components/index.js
export {default as Button} from './Button';


/*
    + FOLDER /src/components/Block/
        + FILE /src/components/Block/index.js
        + FILE /src/components/Block/Block.scss
 */


// FILE /src/components/Block/index.js
import React from "react";
import classNames from "classnames";

import "./Block.scss";

const Block = ({ children, className }) => (
    <div className={classNames('block', className)}>
        {children}
    </div>
);

export default Block;


// FILE /src/components/Block/Block.scss
/*
    .block {
        padding: 45px;
        background: #ffffff;
        box-shadow: 0px 0px 25px rgba(0, 0, 0, 0.4204);
        border-radius: 3px;
    }
 */


// FILE /src/components/index.js
export { default as Button } from './Button';
export { default as Block } from './Block';


// FILE: /src/App.js:
import React, { Component } from "react";
import { Route } from "react-router-dom";

import { Auth } from "pages";

class App extends Component {
    render() {
        return (
            <div className="wrapper">
                <Auth/>
            </div>
        )
    }
}

export default App;


/*
    + FOLDER /src/pages/
        + FOLDER /src/pages/Auth/
            + FILE /src/pages/Auth/index.jsx
            + FILE /src/pages/Auth/Auth.scss

        + FOLDER /src/pages/Home/
            + FILE /src/pages/Home/index.js

        + FILE /src/pages/index.js
 */


// FILE /src/pages/index.js
export { default as Auth } from "./Auth";
export { default as Home } from "./Home";



// FILE /src/pages/Auth/index.jsx
import React from "react";
import { Route } from "react-router-dom";

import { LoginForm, RegisterForm } from "modules";

import "./Auth.scss";

const Auth = () => (
    <section className="auth">
        <div className="auth__content">
            <Route exact path={["/", "/login"]} component={LoginForm} />
            <Route exact path="/register" component={RegisterForm} />
        </div>
    </section>
);

export default Auth;


// FILE /src/pages/Auth/Auth.scss
/*
    .auth {
        display: flex;
        justify-content: center;
        align_items: center;\
        height: 100%;

        &__top {
            text-align: center;
            margin-bottom: 50px;
            h2 {
                font-size: 28px;
                margin-bottom: 5px;
            }
            p {
                font-size: 18px;
                line-height: 21px;
                letter-spacing: 0.1px;
                opacity: 0.5;
            }
        }

        &__register-link {
            text-align: center;
            display: block;
            color: #ADADAD;
            font-size: 16px;
            letter-spacing: 0.1px;
        }

        &__success_block {
            dispay: flex;
            align-items: center;
            justify-content: center;
            flex-direction: column;
            text-align: center;
            height: 400px;
            h3 {
                margin-top: 10px;
            }
            p {
                color: #adadad;
            }
            svg {
                width: 48px;
                height: 48px;
            }
        }

        .button {
            text-transform: uppercase;
            font-weight: 500;
            font-size: 14px;
            letter-spacing: 0.1px;
            weight: 100px;
        }

        .block {
            width: 430px;
        }
    }
*/


// FILE: /src/styles/index.scss
/*
    @import "~antd/dist/antd.css";

    body, html, #root, .wrapper {
        height: 100%;
        color: #202020;
    }
*/

/*
    + FOLDER: /src/modules/
        + FOLDER: /src/modules/LoginForm/
            + FOLDER: /src/modules/LoginForm/containers/
                + FILE: /src/modules/LoginForm/containers/LoginForm.js

            + FOLDER: /src/modules/LoginForm/components/
                + FILE: /src/modules/LoginForm/components/LoginForm.jsx

            + FILE: /src/modules/LoginForm/index.js
 */


// FILE: /src/modules/LoginForm/index.js
import LoginForm from "./containers/LoginForm";

export default LoginForm;


// FILE: /src/modules/LoginForm/components/LoginForm.jsx
import React, {Component} from "react";
import { Form, Icon, Input } from 'antd';
import { Link } from "react-router-dom";

import { Button, Block } from "components";

export default class LoginForm extends Component {
    render() {
        return (
            <div>
                <div className="auth__content">
                    <h2>Войти в аккаунт</h2>
                    <p>Пожалуйста, войдите в свой аккаунт</p>
                </div>
                <Block>
                    <Form onSubmit={this.handleSubmit} className="login-form">
                        <Form.Item>
                            <Input
                                prefix={<Icon type="user" style={{color: 'rgba(0,0,0,.25)'}}/>}
                                placeholder="Username"
                                size="large"
                            />
                        </Form.Item>
                        <Form.Item>
                            <Input
                                prefix={<Icon type="lock" style={{color: 'rgba(0,0,0,.25)'}}/>}
                                type="password"
                                placeholder="Password"
                                size="large"
                            />
                        </Form.Item>
                        <Form.Item>
                            <Button
                                type="primary"
                                size="large"
                            >
                                Войти в аккаунт
                            </Button>
                        </Form.Item>
                        <Link to="/register" className="auth__register-link">
                            Зарегистрироваться
                        </Link>
                    </Form>
                </Block>
            </div>
        )
    }
}

export default LoginForm;


// FILE: /src/modules/LoginForm/containers/LoginForm.js
import LoginForm from "./components/LoginForm";

export default LoginForm;



/*
    $ yarn add react-router
    $ yarn add react-router-dom
*/


/*
    + FILE: /src/modules/index.js
*/

// FILE: /src/modules/index.js
export { default as LoginForm } from "./LoginForm";
export { default as RegisterForm } from "./RegisterForm";


/*
    + FOLDER: /src/modules/RegisterForm/
        + FOLDER: /src/modules/RegisterForm/containers/
            + FILE: /src/modules/RegisterForm/containers/RegisterForm.js

        + FOLDER: /src/modules/RegisterForm/components/
            + FILE: /src/modules/RegisterForm/components/RegisterForm.jsx

        + FILE: /src/modules/RegisterForm/index.js
 */


// FILE: /src/modules/RegisterForm/index.js
import RegisterForm from "./containers/RegisterForm";

export default RegisterForm;


// FILE: /src/modules/RegisterForm/components/RegisterForm.jsx
import React, {Component} from "react";
import { Form, Icon, Input } from 'antd';
import { Link } from "react-router-dom";

import { Button, Block } from "components";

export default class RegisterForm extends Component {
    render() {
        const success = false;

        return (
            <div>
                <div className="auth__content">
                    <h2>Регистрация</h2>
                    <p>Для входа в чат, Вам нужно зарегистрироваться</p>
                </div>
                <Block>
                    { success
                        ?
                        <div className="auth_success-block">
                            <div>
                                <Icon type="info-circle" theme="twoTone" />
                            </div>
                            <h2>Подтвердите свой аккаунт</h2>
                            <p>На Вашу почту отправлено письмо с ссылкой на подтверждение аккаунта.</p>
                        </div>
                        :
                        <Form onSubmit={this.handleSubmit} className="login-form">
                            <Form.Item>
                                <Input
                                    prefix={<Icon type="mail" style={{color: 'rgba(0,0,0,.25)'}}/>}
                                    placeholder="E-mail"
                                    size="large"
                                />
                            </Form.Item>
                            <Form.Item>
                                <Input
                                    prefix={<Icon type="user" style={{color: 'rgba(0,0,0,.25)'}}/>}
                                    type="password"
                                    placeholder="Ваше имя"
                                    size="large"
                                />
                            </Form.Item>
                            <Form.Item>
                                <Input
                                    prefix={<Icon type="lock" style={{color: 'rgba(0,0,0,.25)'}}/>}
                                    type="password"
                                    placeholder="Ваш пароль"
                                    size="large"
                                />
                            </Form.Item>                        <Form.Item>
                            <Input
                                prefix={<Icon type="lock" style={{color: 'rgba(0,0,0,.25)'}}/>}
                                type="password"
                                placeholder="Повторите пароль"
                                size="large"
                            />
                        </Form.Item>
                            <Form.Item>
                                <Button
                                    type="primary"
                                    size="large"
                                >
                                    Зарегистрироваться
                                </Button>
                            </Form.Item>
                            <Link to="/login" className="auth__register-link">
                                Войти в аккаунт
                            </Link>
                        </Form>
                    }
                </Block>
            </div>
        )
    }
}

export default RegisterForm;


// FILE: /src/modules/RegisterForm/containers/RegisterForm.js



/*
    + FILE /src/pages/Home/index.jsx
    + FILE /src/pages/Home/Home.scss
 */


// FILE /src/pages/Home/index.jsx
import React from "react";
import { Route } from "react-router-dom";

import "./Home.scss";

const Home = () => (
  <section className="home">
      <h1>Hellp</h1>
  </section>
);

export default Home;

// 15:15dctv