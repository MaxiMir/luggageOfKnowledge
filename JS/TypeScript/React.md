```shell script
$ npx create-react-app react-typescript --template typescript # --template typescript - флаг, включающий все зависимости для работы с TS

```
### FUNCTIONAL COMPONENT:
```typescript jsx
type HeaderProps = {
	text: string
}

const Header = ({text}: HeaderProps) => <b>{text}</b>
```
```typescript jsx
const HeaderWithoutType: React.FC<{text: string}> = ({text}) => <b>{text}</b>
```
### CLASS:
```typescript jsx
type CounterProps = {
   title?: string
}

type CounterState = {
   count: number
}

class Counter extends Component<CounterProps, CounterState> { // {типизация props}, {типизация state}
	//state = {
	// 	count: 0
	// }

	constructor(props: CounterProps) {
		super(props)

		this.state = {
			count: 0
		}
	}

	static defaultProps: CounterProps = {
		title: "Default title"
	}

	static getDerivedStateFromProps(props: CounterProps, state: CounterState): CounterState | null { // возвращает объект | null для обновления состояния 
		return true ? null : {count: 2}
	}

	componentDidMount(): void {

	}

	shouldComponentUpdate(nextProps: CounterProps, nextState: CounterState): boolean {
		return true
	}

	handleClick = (e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>): void => { // тип события + тип элементов на которые вешаем | React.SyntheticEvent - без уточнения события
        console.log(`${e.clientX}, ${e.clientY}`) 

		this.setState(({count}) => ({
			count: +count
		}))
	}

	render() {
		return (
			<div>
				<p>{this.props.title}{this.state.count}</p>
				<button onClick={this.handleClick}>+1</button>
                <a href="#" onClick={this.handleClick}>Link</a>
			</div>
		)
	}
}
const App: React.FC = () => <Counter title="Counter: "/>

export default App
```
```typescript jsx
class Form extends Comment<{}, {}> {
   handleCopy = (e: React.ClipboadEvent<HTMLInputElement>): void => {
      console.log('Coppied!')
   }

   handleFocus = (e: React.FocusEvent<HTMLInputElement>): void => {
      
   }

   handleSubmit = (e: React.FormEvent<HTMLFormElement>): void => {
      e.preventDefault()
      console.log('Submitted')
   }

   render() {
      return (
         <form onSubmit={this.handleSubmit}>
            <label>
               Simple text:
               <input 
                  type="text"
                  name="text"
                  onCopy={this.handleCopy}
                  onFocus={this.handleFocus}
               />               
            </label>
            <button type="submit">Submit</button>
         </form>
      )  
   }
}

const App: React.FC = () => <Form/>

export default App
```
```typescript jsx
type POSITION = {
   id: string,
   value: string,
   title: string
}

type FormState = {
   inputText: string,
   textareaText: string,
   selectText: string,
   showData: {
      name: string,
      text: string,
      position: string
   }
}

const POSITIONS: Array<POSITION> = [
   {
      id: 'fd',
      value: 'Frontend Developer',
      title: 'Frontend Developer',
   },
   {
      id: 'bd',
      value: 'Backend Developer',
      title: 'Backend Developer',
   }
]

const DEFAULT_SELECT_VALUE: string = POSITIONS[0].value
const styles: React.CSSProperties = {display: 'block', marginBottom: '10px'}

class Form extends Component<{}, FormState> {
   private selectRef = React.createRef<HTMLSelectElement>() // типизация ссылки
   
   state = {
   	  inputText: '',
      textareaText: '',
      selectText: DEFAULT_SELECT_VALUE,
      showData: {
         name: '',
         text: '',
         position: ''
      }  
   }

   handleInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
      const {target: {value: inputText}} = e

      this.setState({inputText})
   }

   handleTextareaChange = (e: React.ChangeEvent<HTMLTextAreaElement>): void => {
      const {target: {value: textareaText}} = e

      this.setState({textareaText})
   }

   handleSelectChange = (e: React.ChangeEvent<HTMLSelectElement>): void => {
      const {target: {value: selectText}} = e

      this.setState({selectText})
   }

   handleShow = (e: React.MouseEvent<HTMLButtonElement>): void => {
      e.preventDefault()

      const {inputText, textareaText, selectText} = this.state
   
      this.setState({
         inputText: '',
         textareaText: '',
         selectText: DEFAULT_SELECT_VALUE,
         showData: {
            name: inputText,
            textareaText: '',
            position: selectText    
         }
      })
   }

   render() {
      const {inputText, textareaText, selectText, showData} = this.state   
      const {name, text, position} = showData

      return (
          <label style={styles}>
             Name:
             <input 
                type="text"
                value={inputText}
                onChange={this.handleInputChange}
             />               
          </label>

          <label style={styles}>
             Text:
             <textarea 
                type="text"
                value={textareaText}
                onChange={this.handleTextareaChange}
             />               
          </label>
         <select 
            style={styles}
            value={selectText}
            onChange={this.handleSelectChange}
            ref={this.selectRef}
         >
         
         </select>
      )  
   }
}
```
### PORTAL (ver. >= 16.3):
Нативный React компонент, который рендерит свое содержимое в любую часть DOM дерева (т.е. вне корневого дива).
Применяется для модальных окон, всплывающих подсказок, тултипов и тд.
```typescript jsx
import React, {Component} from 'react'
import ReactDOM from 'react-dom'

type PortalProps = {
   children: React.ReactNode
}

class Portal extends Component<PortalProps> {
   private el: HTMLDivElement = document.createElement('div')

   public componentDidMount(): void {
      document.body.appendChild(this.el)
   }

   public componentWillUnmount(): void {
      document.body.removeChild(this.el)
   }

   public render(): React.ReactElement<PortalProps> {
      return ReactDOM.createPortal(this.props.children, this.el)
   }
}

/**
 children: JSX.Element # doesn`t account for arrays
 children: JSX.Element | JSX.Element[] # doesn`t accept functions
 children: JSX.ReactChildren # Despite the name, not at all an appropriate type
 children: JSX.ReactChild[] # Better
 children: JSX.ReactNode # Best, accepts everything
*/

class SomeComponent extends Component {
   render() {
      return (
         <div>
            <h1>SomeComponent</h1>
            <Portal>
               <div>TEST PORTAL</div>
            </Portal>
         </div>
      )
   }
}
```
### CONTEXT:
```typescript jsx
interface IContext {
   isAuth: Boolean,
   toggleAuth: () => void
}

// Context creation:
const AuthContext = React.createContext<IContext>({
   isAuth: false,
   toggleAuth: () => {}
})

// Inner component (new syntax of static property):
class Login extends Comment {
   static contextType = AuthContext
   context!: React.ContextType<typeof AuthContext>

   render() {
      const {isAuth, toggleAuth} = this.context
      
      return (
         <button onClick={toggleAuth}>
            {!isAuth ? 'Login' : 'Logout'}
         </button>
      )  
   }
}

// Inner component (old variant with Consumer):
const Profile: React.FC = (): React.React.Element => (
   <AuthContext.Consumer>
      {({isAuth}: IContext) => (
         <h1>{!isAuth ? 'Please login' : 'You are logged in'}</h1>
      )}
   </AuthContext.Consumer>
)

// Root component:
class Context extends Component<{}, {isAuth: Boolean}> {
   readonly state: {
      isAuth: false
   }

   toggleAuth = () => {
      this.setState(({isAuth}) => ({isAuth: !isAuth})
   }

   render() {
      const {isAuth} = this.state
      const context: IContext = {isAuth, toggleAuth: this.toggleAuth}

      return (
         <AuthContext.Provider value={context}>
            <Login/>
            <Profile/>
         </AuthContext.Provider>
      )
   }
}
```
### HOOK useState:
* Inferred as number:
```typescript jsx
const [value, setValue] = useState(0) // при простом значении типизацию делать не надо
```
* Explicitly setting the types:
```typescript jsx
const [value, setValue] = useState<number | undefined>(undefined) // дефолтное значение - undefined
```
```typescript jsx
const [value, setValue] = useState<Array<number>>([]) // дефолтное значение - []
```
```typescript jsx
interface IUser {
   name: sttring,
   age?: number
}

const [value, setValue] = useState<IUser>({name: 'Maxim'})
```
### HOOK useRef:
```typescript jsx
const ref1 = useRef<HTMLElement>(null!) // ref1.current - будет доступен только для чтения и управляемой только через react
const ref2 = useRef<HTMLElement | null>(null) // ref2.current - будет модифицируемым и управляемым мной
```
### HOOK useContext:
```typescript jsx
interface ITheme {
   backgroundColor: string,
   color: string
}

// Context creation:
const ThemeContext = createContext<ITheme>({
   backgroundColor: 'black',
   color: 'white'   
})

// Accessing context in a child component:
const themeContext = useContext<ITheme>(ThemeContext)
```
### HOOK useReducer:
```typescript jsx
interface State {
   count: number
}

type Action = {
   type: 'increment' | 'decrement'
}

const counterReducer = ({count}: State, {type}: Action) => {
   switch (type) {
      case "increment": return {count: count + 1}
      case "decrement": return {count: count - 1}
      default: return {}
   }
}

const [state, dispatch] = useReducer(counterReducer, {count: 0})


dispatch({type: 'increment'}) 
dispatch({type: 'decrement'}) 
```
### HOOK useCallback & useMemo:
* Inferred as number:
```typescript jsx
const memoizedCallback = useCallback(() => {sum(a, b)}, [a, b])
```
* Inferred as (value1: number, value2: number) => number:
```typescript jsx
const memoizedValue = useMemo((a: number, b: number) => {sum(a, b)}, [a, b])
```
### HOOK useEffect & useLayoutEffect:
```typescript jsx
useEffect(() => {
   const subscriber = subsribe(options)

  return () => {
    unsubscribe(subscriber)
  };
}, [options]);
```
### HOC:
```typescript jsx
type BaseProps = {
   primTitle: string,
   secTitle?: string
}

type InjectedProps = {
   toggleStatus: Boolean,
   toggle: () => void
}

const Button = ({primTitle, secTitle, toggle, toggleStatus}: any) => (
   <button onClick={toggle}>
      {toggleStatus ? primTitle : secTitle}
   </button>
)

const withToggle = <BaseProps extends InjectedProps>(PassedComponent: React.ComponentType<BaseProps>) => {
   return (props: BaseProps) => {
      const [toggleStatus, toggle] = useState(false)
    
      return (
         <PassedComponent
            {...props as BaseProps}
            toggle={() => toggle(!toggleStatus)}
            toggleStatus={toggleStatus}   
         />
      )  
   }
}

const ToggleButton = withToggle(Button)

const App: React.FC = () => <ToggleButton primTitle="Main Title" secTitle="Additional Title"/>

export default App
```
```typescript jsx
interface WithLoadingProps {
   loading: boolean
}

const withLoading = <P extends object>(Component: React.ComponentType<P>) => {
    class WithLoading extends React.Component<P & WithLoadingProps> {
	  render() {
	     const {loading, ...props} = this.props

         return loading ? <LoadingSpinner/> : <Component {...props as P}/>
      }
    }
}
```
### HTTP RESPONSE:
```ts
interface HttpResponse<T> extends Response {
	parsedBody?: T
	status: number,
	redirect: boolean
	// ... and other necessary parameters
}

export async function http<T>(request: string): Promise<HttpResponse<T>> {
	const response: HttpResponse<T> = await fetch(request)
	
   try {
      response.parsedBody = await response.json()
   } catch (ex) {}
   
   if (!response.ok) {
      // Error if there is response status issue
      throw new Error(response.statusText)
   }
	
   return response
}

interface IPost {
	title?: string,
	body?: string,
}

try {
   const resp = await http<IPost>('https://jsonplaceholder.typicode.com/posts/')
} catch (resp) {
   console.error('Error', resp)
}
```
