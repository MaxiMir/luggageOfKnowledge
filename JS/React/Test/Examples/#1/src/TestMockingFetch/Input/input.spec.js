import React from "react";
import Input from "./input";

describe("Input component", () => {
  it("should render Input component", () => {
    const component = shallow(<Input />);
    expect(component).toMatchSnapshot();
  });

  it("should call onClick method", () => {
    const mockCallBack = jest.fn();
    const component = shallow(<Input onChange={mockCallBack} />);
    expect(mockCallBack.mock.calls.length).toBe(0);
    component.find(".input").simulate("change");
    expect(mockCallBack.mock.calls.length).toBe(1);
  });

  describe("defaultProps", () => {
    it("should use default onChange", () => {
      const result = Input.defaultProps.onChange();
      expect(result).toBe(undefined);
    });

    it("should use default onKeyPress", () => {
      const result = Input.defaultProps.onKeyPress();
      expect(result).toBe(undefined);
    });
  });
});
