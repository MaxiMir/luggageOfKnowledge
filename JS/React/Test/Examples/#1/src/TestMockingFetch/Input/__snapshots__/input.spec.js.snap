// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render Input component 1`] = `
<div
  className="inputWrapper"
>
  <i
    className="fas fa-search"
  />
  <input
    className="input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Click to search"
    value=""
  />
</div>
`;
